doc/radosgw/multisite.rst

```diff
@@ -0,0 +1,487 @@
+
+- **Realm**: A realm is a container for zonegroups, this allows for separation
+  of zonegroups themselves between clusters. It is possible to create multiple
+  realm, making it easier to run completely different configurations in the same
+  cluster.
+
+- **Period**: A period holds the configuration structure for the current state
+  of the realm, a period needs to be updated whenever there is a zone
+  configuration change of the master zone.
+
+
+
+Pool Configuration
+==================
+
+About this guide
+================
+
+In this guide we create a single zone group, with two seperate zones, which
+actively sync data between them. There is no sync agent involved for mirroring
+data changes between the RadosGWs and this allows for a much simpler
+configuration scheme and active-active configurations. Please note that metadata
+operations such as creating a user would still need to go through the master
+zone, however data operations such as creation of buckets objects etc. can be
+handled by any of the zones.
+
+System Keys
+===========
+
+While configuring zones, RadosGW, expects creation of zone a system user, with
+an S3 like access and secret keys. This allows another radosgw instance to pull
+the configuration remotely, with the access and secret keys. For making
+scripting and use of configuration management tools easier, it would be easier
+to generate the access-key and secret-key before hand and use these. For the
+purpose of the guide we assume an access key and secret key are set in the
+environment. For eg., something like::
+
+  $ SYSTEM_ACCESS_KEY=1555b35654ad1656d805
+  $ SYSTEM_SECRET_KEY=h7GhxuBLTrlhVUyxSPUKUV8r/2EI4ngqJxD7iBdBYLhwluN30JaT3Q==
+
+Naming conventions
+==================
+
+This section describes the process of setting up a master zone. For the purpose
+of the guide, we'll assume a zonegroup called ``us`` spanning the United States,
+which will be our master zonegroup. This will contain two zones written in a
+``{zonegroup}-{zone}`` format, this is just a convention only and you are free
+to choose a format you prefer to set these names. So in summary:
+
+- Master Zonegroup: United States ``us``
+- Master zone: United States, East Region: ``us-east``
+- Secondary zone: United States, West Region: ``us-west``
+
+This will be a part of a larger realm, say ``gold``
+
+Creating a realm
+----------------
+
+Configure a realm called ``gold``, which for ::
+
+  # radosgw-admin realm create --rgw-realm=gold
+  {
+    "id": "4a367026-bd8f-40ee-b486-8212482ddcd7",
+    "name": "gold",
+    "current_period": "09559832-67a4-4101-8b3f-10dfcd6b2707",
+    "epoch": 1
+  }
+
+
+Note that every realm has an id, which allows for flexibility like renaming the
+realm later, should the need arise. The ``current_period`` changes whenever we
+change anything in the master zone. The ``epoch`` is incremented when there's a
+change in configuration which doesn't modify the master zone
```

Comment:
> The ``current_period`` changes whenever we change anything in the master zone. The ``epoch`` is incremented when there's a change in configuration which doesn't modify the master zone

this is a bit off, from my understanding. any change to the configuration that doesn't change which zone is the master_zone will just increment the _period's_ epoch. on the other hand, changing the master_zone to point to a different zone will a) generate a new period id with period epoch=1, b) assign realm's current_period to the new period's id, and c) increment the realm's epoch. be careful not to confuse the realm's epoch with the period's epoch